app/context/app/appContext.js

```diff
@@ -1,15 +1,26 @@
+import PropTypes from 'prop-types'
 import React, { createContext, useContext, useReducer } from 'react'
 import initialState from './state'
 import reducer from './reducer'
 
 export const AppContext = createContext()
 
-export function AppStateProvider(props) {
+export function AppStateProvider({ children }) {
```

Comment:
that's what I prefer to do but I think there should be a guideline cause using arrow functions all way may cause hoisting related issues later on, there can be simple rules to prevent

Reply:
The Fat Arrow functions or JavaScript Arrow functions prevent the `this` bug. because they have their own scope without declaring `this` object. this ability helps us to make a big collaborate between components. there are too many articles to explain this usage. for example please take a look at [this article](https://frontarm.com/james-k-nelson/when-to-use-arrow-functions/). But if, after searching or reading you decide to write the stateless components as a legacy function declaration I respect your decision.

Reply:
As I know using arrows may cause hoisting related problem sometimes but possible to prevent! so let's do arrow function in whole project man \m/

Reply:
🌹